pkg/apis/work/v1alpha1/well_known_constants.go

```diff
@@ -16,19 +16,25 @@ limitations under the License.
 
 package v1alpha1
 
+// TODO(whitewindmills): These deprecated labels will be removed in a future version.
 const (
 	// ResourceBindingNamespaceLabel is added to objects to specify associated ResourceBinding's namespace.
+	// Deprecated
 	ResourceBindingNamespaceLabel = "resourcebinding.karmada.io/namespace"
 
 	// ResourceBindingNameLabel is added to objects to specify associated ResourceBinding's name.
+	// Deprecated
 	ResourceBindingNameLabel = "resourcebinding.karmada.io/name"
 
 	// ClusterResourceBindingLabel is added to objects to specify associated ClusterResourceBinding.
+	// Deprecated
 	ClusterResourceBindingLabel = "clusterresourcebinding.karmada.io/name"
 
 	// WorkNamespaceLabel is added to objects to specify associated Work's namespace.
+	// Deprecated
```

Comment:
Can work-related labels be split into a separate pr?

Reply:
In order to process PRs in parallel, this PR just marks work-related labels as deprecated, handling work-related labels will be in https://github.com/karmada-io/karmada/pull/4813.

Reply:
I tend to don't touch these labels, let this PR focus on resourcebinding.karmada.io/key and clusterresourcebinding.karmada.io/key. But I'm ok with it.

Reply:
fine, let's focus on rb/crb.